src/client/Client.cc

```diff
@@ -12422,8 +12422,6 @@ int Client::ll_file_layout(Fh *fh, file_layout_t *layout)
 int Client::ll_get_stripe_osd(Inode *in, uint64_t blockno,
 			      file_layout_t* layout)
 {
-  Mutex::Locker lock(client_lock);
-
   inodeno_t ino = ll_get_inodeno(in);
```

Comment:
I think this can get directly from `in`. `in->ino`.

Reply:
In the case, no one is calling this function ll_get_inodeno ( and consecutively) _get_inodeeno except ll_get_stripe_osd. May be we can remove these functions?

Reply:
`Client::_ll_forget()` is calling `_get_inodeno()`

Reply:
@joscollin in _ll_forget as well we can have inodeno_t ino = in->ino. But I am not aware of design behind using _get_inodeno(). 

Besides that, I see in some the functions in Client.cc, we have taken client_lock but don't unlock it before leaving the function. For example, ll_get_stripe_osd, ll_readlink, ll_get_snapid. This seems like error. If that's case I can submit a another patch.

Reply:
See how `Mutex::Locker` is defined.

Reply:
RAII